components/dom_struct/lib.rs

```diff
@@ -46,6 +46,14 @@ pub fn dom_struct(args: TokenStream, input: TokenStream) -> TokenStream {
                         &self.#ident
                     }
                 }
+
+                impl Eq for #name {}
+
+                impl PartialEq for #name {
+                    fn eq(&self, other: &Self) -> bool {
+                        crate::DomObject::reflector(self) == crate::DomObject::reflector(other)
+                    }
+                }
```

Comment:
I think this should be done in domobject_derive, somewhere around here: https://github.com/servo/servo/blob/590527176e7c03648fcce797604d3e12da0c2c99/components/domobject_derive/lib.rs#L48